liquibase-standard/src/main/java/liquibase/resource/AbstractResource.java

```diff
@@ -13,16 +16,51 @@ public abstract class AbstractResource implements Resource {
 
     public AbstractResource(String path, URI uri) {
         this.originalPath = path;
-        this.path = path
-                .replace("\\", "/")
-                .replaceFirst("^classpath\\*?:", "")
-                .replaceFirst("^/", "");
-
         if (uri != null) {
-            this.uri = uri.normalize();
+            this.uri = handleUriSyntax(uri);
         } else {
             this.uri = null;
         }
+
+        String pathL = path
```

Comment:
Can you come up with a more descriptive variable name than `pathL`? It would be easy to accidentally type `path` and not notice.